core/types/printing.cc

```diff
@@ -154,8 +154,114 @@ string AndType::toStringWithTabs(const GlobalState &gs, int tabs) const {
                        rightBrace ? ")" : "");
 }
 
+/**
+ * Metadata collected while traversing AndType for pretty-printing.
+ */
+struct AndInfo {
```

Comment:
All of this is essentially a copy&paste of the `OrInfo` case (and all of its methods).

We are not using all of the functionality for `And`, so we could clean it up, but not sure how much I like the idea of making them diverge too much.

We could also think about just renaming the `Or` case and use it for both `Or` and `And`

Reply:
I think it would be good to remove the redundant info, and specialize this structure for pretty-printing And.

Reply:
👍🏽 Looked closely into it, and the only thing that is being used is `containsMultiple`, so I removed all the rest

Reply:
Actually, is `AndInfo` even necessary? Won't there always be multiple entries in the result?